### PR TITLE
Add `session` authentication service

### PR DESCRIPTION
**Pull request by patocallaghan**
Chatted with @mansona and @Alonski about pushing things forward, we decided to simplify things by removing ember-simple-auth.

I've brought over the `session` service which I added to my [Realworld Octane branch](https://github.com/patocallaghan/realworld-starter-kit/blob/4611cda2c1b9f391f976b111e2e23a833d3dd00c/app/services/session.js).

This PR does the following:

* Removes ember-simple-auth
* Bump ember-qunit
* Remove lots of "empty" route unit tests
* Adds a session service which can register a user, as well as log them in and out, and stores the `token` in local storage.
  * `session#register(username, email, password)`
  * `session#logIn(email, password)`
  * `session#logOut()`
  * `session#isLoggedIn`

### CODE REVIEW

File: app/controllers/login.js [hunk 1]
```diff
@@ -1,30 +1,19 @@
...
-    login(email, password) {
-      return this.session
-        .authenticate('authenticator:conduit', { email, password })
-        .then(() => {
-          this.transitionToRoute('home');
-        })
-        .catch(normalizedErrors => {
-          set(this, 'errors', normalizedErrors);
-        });
+    async logIn() {
```
Comment (mansona) [inline app/controllers/login.js]: async actions can have a whole load of gotyas (especially in testing) so I would strongly advise against using them 🙈 it is also harder to disable the login button while you are logging in (preventing the double click issue)

If you make this an ember concurrency task you will be able to save issues with testing and easily disable the button while the task is in progress

Comment (patocallaghan) [reply app/controllers/login.js]: I thought all of that runloop/testing stuff was resolved when Ember shipped the Microtasks integration in 3.4? I'm going off of the very long discussion in [this Discuss post](https://discuss.emberjs.com/t/readers-questions-why-does-ember-still-use-rsvp/14736/36)

I do get the point about disabling the button though. I'm trying to remember our conversation about dependencies. We've decided it's okay to use ember-concurrency?

Comment (mansona) [reply app/controllers/login.js]: yes we said that if we're doing any async actions then we should be using ember-concurrency 👍

Comment (Alonski) [reply app/controllers/login.js]: I agree with @mansona here :)
Let's use `ember-concurrency` where appropriate?

Comment (patocallaghan) [reply app/controllers/login.js]: okely dokely 👍 will update. Thank!

Comment (patocallaghan) [reply app/controllers/login.js]: @Alonski @mansona this should be good for another 👀 now

**Comment by mansona** [comment]
@Alonski it looks like I don't have merge permissions on this repo 🤔 I'm not sure I need them but I don't know if my reviews count for anything if I don't

**Comment by Alonski** [comment]
@mansona Your review definitely counts :)
I will merge this in a few hours once work calms down a bit 😄

**Comment by patocallaghan** [comment]
Ah I can't merge either 😬 Could I get merge rights too please whenever you get the chance @Alonski?

**Comment by Alonski** [comment]
Closes #36 #37

**Pull request merged**